chall05/mtuomine.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+#ruby 2.5.1p57
+
+require 'net/http'
+
+uri = URI("https://en.wikipedia.org/wiki/" + ARGV[0])
+counter = 0
+visited = []
+
+def traverse(visited, uri, counter)
+	html = Net::HTTP.get(uri)
+	if html.empty?
+		puts "no wikipedia page for this word!"
+		exit
+	end
+
+	content = html.split("<p>", 2)
+	linksArr = content[1].scan(/href\s*=\s*"([^"]*)"/)
+
+	# Bubblegum, couldn't combine with regex on tpo
+	res = []
+	linksArr.each { |link|
+		if link[0].match(/^\/wiki\//)
+			res.append(link[0])
+		end
+	}
+
+	res.each { |link|
+		trimmed = link.split('/').last
+		if trimmed.eql? "Philosophy"
+			puts "!!! Reach Philosophy !!!"
+			exit
+		elsif visited.include? trimmed
```

Comment:
Ruby allows this kind of argument passed but I will always suggest keeping writing 
```ruby
visited.include?(trimmed)
```

To keep consistent with other languages you develop or will develop with.